### PR TITLE
Add reminder to project

### LINKED ISSUE #71: As a user, I can add a reminder to a project.

### LINKED ISSUE #73: Add a reminder to a project

### PR DESCRIPTION
**Pull request by claracheong4**
Refactoring:
Instance -> InstanceWrapper
DateTime -> DateTimeWrapper
Date -> DateWrapper
Time -> TimeWrapper
Timer -> TimerWrapper

Added a JsonToModel class to convert json fields to model fields, the class is use in JsonAdaptedProject and JsonAdaptedTask

Added a ReminderManager to handle scheduling of reminders. 
Added a ReminderCommand to hide the reminder pane.

### CODE REVIEW

File: src/main/java/seedu/momentum/logic/parser/ShowComponentCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            return name().charAt(0) + name().substring(1).toLowerCase();
+        }
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ShowComponentCommand
+     * and returns an ShowComponentCommand object for execution.
+     *
+     * @param model the current model.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (pr4aveen) [inline src/main/java/seedu/momentum/logic/parser/ShowComponentCommandParser.java]: . after `@throws'

File: src/main/java/seedu/momentum/model/project/Deadline.java [hunk 2]
```diff
@@ -25,62 +25,62 @@
...
-    public Deadline(String date, Date createdDate) {
-        requireNonNull(date);
-        requireNonNull(createdDate);
-        checkArgument(Date.isValid(date), Date.MESSAGE_CONSTRAINTS);
-        checkArgument(!isBeforeCreatedDate(date, createdDate), CREATED_DATE_MESSAGE_CONSTRAINT);
-        this.date = Optional.of(new Date(date));
-        this.time = Optional.empty();
+    public Deadline(String dateWrapper, DateWrapper createdDateWrapper) {
+        requireNonNull(dateWrapper);
+        requireNonNull(createdDateWrapper);
```
Comment (pr4aveen) [inline src/main/java/seedu/momentum/model/project/Deadline.java]: I think `requireAllNonNull(date, createdDate)` can be used here and in the constructor below.

File: src/main/java/seedu/momentum/model/project/Task.java [hunk 3]
```diff
@@ -20,57 +21,72 @@
...
         UniqueDurationList newDurations = new UniqueDurationList();
         newDurations.setDurations(durations);
         newDurations.add(duration);
-        return new Task(name, description, completionStatus, createdDate, deadline, tags, newDurations, newTimer);
+        return new Task(name, description, completionStatus, createdDateWrapper, deadline, reminder, tags,
+                newDurations, newTimerWrapper);
+    }
+
+    /**
+     * Remove the reminder of a project.
```
Comment (pr4aveen) [inline src/main/java/seedu/momentum/model/project/Task.java]: Replace `Remove` with `Removes`

**Review by pr4aveen — APPROVED** [review]
Just some minor javadoc comments. LGTM otherwise 👍

**Review by kkangs0226 — APPROVED** [review]
LGTM

**Pull request merged**